### PR TITLE
GEOPY-1645: Update for geoh5py refactor of class instantiation

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-1645 - Update for geoh5py refactor of class instantiation**

### CODE REVIEW

File: geoh5py/objects/curve.py [hunk 1]
```diff
@@ -173,7 +173,7 @@ def validate_cells(self, indices: list | tuple | np.ndarray | None) -> np.ndarra
         if not np.issubdtype(indices.dtype, np.integer):
             raise TypeError("Indices array must be of integer type")
 
-        if np.max(indices) > self.n_vertices - 1:
+        if len(indices) > 1 and np.max(indices) > (self.n_vertices - 1):
```
Comment (MatthieuCMira) [inline geoh5py/objects/curve.py:176]: I don't get it, len(indices) will always be >1 no?
Don't you mean <=1?

**Review by MatthieuCMira — COMMENTED** [review]
Seems good to me. 
Just one comment. I'm not sure I'm understanding a line

File: geoh5py/objects/curve.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/objects/curve.py:176]: It's possible that all vertices get removed, which crashes the np.max()

**Review by MatthieuCMira — DISMISSED** [review]
ok my bad I did not understood

**Review by benk-mira — APPROVED** [review]
yup!

**Pull request merged**